src/buildercore/core.py

```diff
@@ -248,11 +252,16 @@ def _ec2_connection_params(stackname, username, **kwargs):
     return params
 
 @contextmanager
-def stack_conn(stackname, username=config.DEPLOY_USER, **kwargs):
+def stack_conn(stackname, username=config.DEPLOY_USER, node=None, **kwargs):
     if 'user' in kwargs:
         LOG.warn("found key 'user' in given kwargs - did you mean 'username' ??")
 
-    data = stack_data(stackname, ensure_single_instance=True)[0]
+    data = stack_data(stackname, ensure_single_instance=False)
+    ensure(len(data) == 1 or node, "stack is clustered and no node specified")
+    if node:
+        ensure(utils.isint(node) and int(node) > 0, "given node must be an integer and greater than zero")
+        node = int(node) - 1 # decrement to a zero-based value
```

Comment:
would not overwrite a method parameter with a value carrying a different meaning, `data_index` or `node_index` could be created with a default of `0`

Reply:
this is a function, not a method. I can rename 'node' during decrement to something else